src/Extrinsics.js

```diff
@@ -1,22 +1,21 @@
 import React, { useEffect, useState } from "react";
-
 import { Grid, Form, Dropdown, Input } from "semantic-ui-react";
 
-import TxButton from "./TxButton";
+import { useSubstrate } from "./substrate";
+import { TxButton } from "./substrate/components";
 
 export default function Extrinsics(props) {
-  const { api, accountPair } = props;
-
+  const { api } = useSubstrate();
   const [modulesList, setModulesList] = useState([]);
-  const [status, setStatus] = useState("");
+  const [status, setStatus] = useState(null);
   const [callableFunctionList, setCallableFunctionList] = useState([]);
+  const { accountPair } = props;
```

Comment:
It super nice to not pass the api all over the place through the component tree. I wonder if we could do the same with the account pair? I don't know how feasible that would be.

Reply:
I have thought about it when refactoring. But when it comes to selecting an account, it is really more of a choice of the app developer. It is something he should handle. 

My intent is to build up a library layer so app dev don't need to touch, or understand, the internal code. I also exposed out `apiState` and `keyringState`. If the refactor is good, I will need to document those states.